purchase_order_variant_mgmt/wizard/purchase_manage_variant.py

```diff
@@ -55,20 +60,23 @@ def _onchange_product_tmpl_id(self):
                 order_line = purchase_order.order_line.filtered(
                     lambda x: x.product_id == product
                 )[:1]
-                lines.append(
-                    (
-                        0,
-                        0,
-                        {
-                            "value_x": value_x,
-                            "value_y": value_y,
-                            "product_uom_qty": order_line.product_qty,
-                        },
+                try:
+                    lines.append(
+                        (
+                            0,
+                            0,
+                            {
+                                "value_x": value_x.id,
+                                "value_y": value_y.id,
+                                "product_uom_qty": order_line.product_qty,
+                            },
+                        )
                     )
-                )
+                except Exception:
+                    _logger.info("Product '%s' doesn't have any variant", template.name)
+                    continue
```

Comment:
@sonhd91 which context did you face with the exception? Besides, hiding the exception by putting it in the logfile is not clear to the end user

Reply:
We got exceptions when `template` just have 1 lines attribute >> attr_lines = 1

Reply:
are the test cases passed without handling the exception?

Reply:
test cases all pass without exception